From 794aa125a3bb019f5b47ce8242e5d189b4404994 Mon Sep 17 00:00:00 2001
From: tpg <phenomenal@wp.pl>
Date: Sat, 7 Sep 2013 20:15:36 +0000
Subject: [PATCH] - handle /etc/locale.conf - by default this file should
 contain same configuration as /etc/sysconfig/i18n - used by systemd's
 loginctl

---
 perl-install/any.pm  | 1 +
 perl-install/lang.pm | 6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/perl-install/any.pm b/perl-install/any.pm
index 15f93bc..c0b22a9 100644
--- a/perl-install/any.pm
+++ b/perl-install/any.pm
@@ -1328,6 +1328,7 @@ sub report_bug {
       header("urpmi.cfg"), cat_("$::prefix/etc/urpmi/urpmi.cfg"),
       header("modprobe.preload"), cat_("$::prefix/etc/modprobe.preload"),
       header("sysconfig/i18n"), cat_("$::prefix/etc/sysconfig/i18n"),
+      header("locale.conf"), cat_("$::prefix/etc/locale.conf"),
       header("/proc/iomem"), cat_("/proc/iomem"),
       header("/proc/ioport"), cat_("/proc/ioports"),
       map_index { even($::i) ? header($_) : $_ } @other;
diff --git a/perl-install/lang.pm b/perl-install/lang.pm
index a255abc..b517691 100644
--- a/perl-install/lang.pm
+++ b/perl-install/lang.pm
@@ -1081,8 +1081,8 @@ sub lang_changed {
 
 sub read {
     my ($b_user_only) = @_;
-    my ($f1, $f2) = ("$::prefix$ENV{HOME}/.i18n", "$::prefix/etc/sysconfig/i18n");
-    my %h = getVarsFromSh($b_user_only && -e $f1 ? $f1 : $f2);
+    my ($f1, $f2, $f3) = ("$::prefix$ENV{HOME}/.i18n", "$::prefix/etc/sysconfig/i18n", "$::prefix/etc/locale.conf");
+    my %h = getVarsFromSh($b_user_only && -e $f1 ? $f1 : $f2 && $f3);
     my $locale = system_locales_to_ourlocale($h{LC_MESSAGES} || 'en_US', $h{LC_MONETARY} || 'en_US');
     
     if (find { $h{$_} } @IM_i18n_fields) {
@@ -1188,7 +1188,7 @@ sub write {
         add2hash($h, { CONSOLE_NOT_LOCALIZED => 'yes' });
     }
 
-    my $file = $b_user_only ? "$ENV{HOME}/.i18n" : '/etc/sysconfig/i18n';
+    my $file = $b_user_only ? "$ENV{HOME}/.i18n" : '/etc/sysconfig/i18n' && '/etc/locale.conf';
     log::explanations(qq(Setting l10n configuration in "$file"));
     setVarsInShMode($::prefix . $file, 0644, $h);
 
--- drakxtools-14.41/perl-install/standalone/finish-install.xsetup.orig	2013-10-29 10:18:27.129378345 +0000
+++ drakxtools-14.41/perl-install/standalone/finish-install.xsetup	2013-10-29 10:19:16.033380917 +0000
@@ -5,8 +5,8 @@
 if [ -f /etc/sysconfig/finish-install -a -x /usr/sbin/finish-install ]; then
     . /etc/sysconfig/finish-install
     if [ "$FINISH_INSTALL" = yes ]; then
-	if [ -r /etc/sysconfig/i18n ]; then
-	    . /etc/sysconfig/i18n
+	if [ -r /etc/locale.conf ]; then
+	    . /etc/locale.conf
 	    export LANGUAGE LC_ADDRESS LC_COLLATE LC_NAME LC_NUMERIC LC_MEASUREMENT LC_TIME \
 	    LANG LC_IDENTIFICATION LC_MESSAGES LC_CTYPE LC_TELEPHONE LC_MONETARY LC_PAPER
 	fi
